src/policy/mempool_fees.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024 The Bitcoin Core developers
+// Distributed under the MIT software license. See the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <logging.h>
+#include <node/miner.h>
+#include <policy/estimator.h>
+#include <policy/mempool_fees.h>
+#include <policy/policy.h>
+#include <util/trace.h>
+
+
+using node::GetNextBlockFeeRateAndVsize;
+
+ForecastResult MemPoolPolicyEstimator::EstimateFee(unsigned int targetBlocks)
+{
+    return EstimateFeeWithMemPool(targetBlocks);
+}
+
+ForecastResult MemPoolPolicyEstimator::EstimateFeeWithMemPool(unsigned int targetBlocks)
+{
+    if (targetBlocks > MEMPOOL_FORECAST_MAX_TARGET) {
+        return ForecastResult(strprintf("%s: Confirmation target %s is above maximum limit of %s, mempool conditions might change and forecast above %s block target are unreliable",
+                                        MEMPOOL_FORECAST_NAME_STR, targetBlocks, MEMPOOL_FORECAST_MAX_TARGET, MEMPOOL_FORECAST_MAX_TARGET));
+    }
+
+    if (!m_mempool->GetLoadTried()) {
+        return ForecastResult(strprintf("%s: Mempool did not finished loading, can't get accurate fee rate forecast", MEMPOOL_FORECAST_NAME_STR));
```

Comment:
In  [policy, fees]: create a mempool based fee rate forecaster:

```suggestion
        return ForecastResult(strprintf("%s: Mempool not finished loading; can't get accurate fee rate forecast", MEMPOOL_FORECAST_NAME_STR));
```